File: 2093-check-if-string-is-a-prefix-of-array/check-if-string-is-a-prefix-of-array.cpp
class Solution {
public:
    bool isPrefixString(string s, vector<string>& words) {
        string sub = "";

        for (string str : words) {
            for (char ch : str) {
                sub += ch;
            }
            if (sub == s) {
                return true;
            }
        }

        return false;
    }
};

File: 2093-check-if-string-is-a-prefix-of-array/check-if-string-is-a-prefix-of-array.js
/**
 * @param {string} s
 * @param {string[]} words
 * @return {boolean}
 */
var isPrefixString = function (s, words) {
    let i = 0;

    for (const word of words) {
        if (!s.startsWith(word, i)) {
            return false;
        }
        i += word.length;
        if (i == s.length) {
            return true;
        }
    }
    return false;
};